project/src/store/api-actions.js

```diff
@@ -0,0 +1,9 @@
+import {ActionCreator} from './action';
+import {APIRoute} from '../const';
+import {adaptToClient} from './adapter';
+
+export const fetchFilms = () => (dispatch, _getState, api) => (
+  api.get(APIRoute.FILMS)
+    .then(({data}) => data.map((film) => adaptToClient(film)))
+    .then((films) => dispatch(ActionCreator.loadFilms(films)))
+);
```

Comment:
```suggestion
    export const fetchFilms = () => async (dispatch, _getState, api) => {
      const {data} = await api.get(APIRoute.FILMS);
      const films = data.map((film) => adaptToClient(film));
      dispatch(ActionCreator.loadFilms(films));
    };
```

Мудро постуаешь, заказывая списко фильмов перед началом работа приложения.

А вот как думаешь, если бы заказчик попросил "чтобы список сам обновлялся по мере появления новых фильмов". Как бы ты поступил?